### ISSUE TITLE
Supabase Problems

### ISSUE THREAD

**Issue by Triggsmt67** [stackblitz/bolt.new#4455]
### Describe the bug

My project was working fine until Bolt.new forced me to create a new data base.  I do not want to keep burning tokens to fix the issue.  I used 3million tokens fixit the first time. I believe the problem is the way the SQL  created the table.This is the problem bolt created : Failed to run sql query: ERROR:  428C9: cannot insert a non-DEFAULT value into column "email"
DETAIL:  Column "email" is a generated column.
CONTEXT:  SQL statement "insert into auth.identities (
      id,
      user_id,
      identity_data,
      provider,
      provider_id,
      created_at,
      updated_at,
      email
    ) values (
      new_user_id,
      new_user_id,
      format('{"sub":"%s","email":"%s"}', new_user_id::text, 'triggsmt67@gmail.com')::jsonb,
      'email',
      'triggsmt67@gmail.com',
      now(),
      now(),
      'triggsmt67@gmail.com'
    )"
PL/pgSQL function inline_code_block line 33 at SQL statement

Second Error: Failed to run sql query: ERROR:  23505: duplicate key value violates unique constraint "schema_migrations_pkey"
DETAIL:  Key (version)=(0002) already exists.

### Link to the Bolt URL that caused the error

https://bolt.new/~/sb1-cu3m4wq7

### Validations

- [X] Please make your project public or accessible by URL. This will allow anyone trying to help you to easily reproduce the issue and provide assistance.

### Steps to reproduce

I can not ad Admin Login to Supabase

### Expected behavior

You enter your email and Password and then your directed to the admin page
<img width="522" alt="Screenshot 2024-12-19 at 2 55 58 PM" src="https://github.com/user-attachments/assets/5b702e50-af76-4304-87af-15b1589ef43d" />

### Platform

```
Browser name = Chrome
Full version = 131.0.0.0
Major version = 131
navigator.appName = Netscape
navigator.userAgent = Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36
performance.memory = {
  "totalJSHeapSize": 69211415,
  "usedJSHeapSize": 66111375,
  "jsHeapSizeLimit": 4294705152
}
Username = Triggsmt67
Chat ID = 6d5b05a9d4bb
```

**Comment by sdrprod** [comment]
Following. Stuck on this exact issue . 
![image](https://github.com/user-attachments/assets/186a9d91-78e1-4dd4-87b3-3fce2b5d7a3d)

**Comment by Triggsmt67** [comment]
This is the problem, prior to bolt migrating to a new database I had solved these problems. I think I might just need to start over on a new project, I am not good at coding and there are a lot of issues to fix

**Comment by Azraelblame** [comment]
yeah I'm also having the same issue. just burned through 4 mil tokens. ridiculous

**Comment by mrtawil** [comment]
+1

**Comment by Azraelblame** [comment]
So I stopped using Supabase in bolt. My advise, don’t use supabase in bolt, it’s a headache. It’s not properly implemented. I have reverted back to using firebase by just creating the database manually myself. You can ask bolt to give you the instructions on what you might need but what I can tell you is this, either this tool is designed to periodically delete files on purpose or it’s just garbage. And what has been really frustrating from me is that it knows exact What it removed when you just ask it why it removed files even without mentioning which files. Either that or these no code tools still have a long way. And what I can tell you is, if you do not know any code and you want to build something a little more complex than just a landing page then you better  start learning otherwise if you think these tools will do everything then you are in for it. I’ve basically just told myself whatever money I’m burning right now is for learning. Bolt gets expensive real quick once you start hitting problems with your backend. My advise don’t let bolt try to fix them, copy the entire problem and paste into chatgpt, chatgpt will explain what is going and give you some solutions and then you copy that response and paste into bolt 9/10 it will actually fix that issue at the first go. (At least lovable fixes issues for free but it’s also another heachache once you start dealing with backends even though their supabase integration is much better) and if you are doing something more than just a front UI, even with that even, you need to explicitly instruct it to NOT CHANGE ANYTHING OR ALTER ANY CODE EXCEPT FIXING THIS PARTICULAR PROBLEM. Add this in every message and if it creates something you like a design or a function is working well you have to instruct it not to change it, in every message going forward. I know it sucks but unless you want a whole library removed and you burn through tokens, like it’s Christmas, well it is Christmas (Merry Christmas) her Bolt.newness will become a prophet and in her infinite wisdom will just decide you don’t need a whole file and just remove it. I know this was long but I’ll do another message with images when I get time and also give my tips on what I have found works with this, most of the times, annoying AI

**Comment by ian-tyro** [comment]
+1

**Comment by stevenestnook** [comment]
lost so many tokens to this! Feel like i've been robbed, you go ahead and manually sort the migration files and bolt undoes your work. It says it's found the issue and has renamed the files and tries again, all the while the files are still the old file names/keys .... Trying the firebase recommendation for now until this is sorted.

**Comment by Azraelblame** [comment]
> lost so many tokens to this! Feel like i've been robbed, you go ahead and manually sort the migration files and bolt undoes your work. It says it's found the issue and has renamed the files and tries again, all the while the files are still the old file names/keys .... Trying the firebase recommendation for now until this is sorted.

Try windsurf. It’s also got it’s issues but it way better. And if it effs up you can always click a message (revert) to when things were working and it will just load the working version you and you can start from there. Also forking your periodically helps too.
